pandas/io/common.py

```diff
@@ -504,12 +466,8 @@ def infer_compression(
     ------
     ValueError on invalid compression specified.
     """
-    # No compression has been explicitly specified
-    if compression is None:
-        return None
-
     # Infer compression
-    if compression == "infer":
+    if compression in ("infer", None):
```

Comment:
this is for consistency across `to/read_` functions: `read_fwf` uses `None` to infer the compression.

https://github.com/pandas-dev/pandas/blob/8b05fe3bd20965ba64477f6c96dbf674d9f155ff/pandas/tests/io/parser/test_read_fwf.py#L597